### PR TITLE
Don't deploy docs on master

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -174,24 +174,12 @@ workflows:
       - docs:
           requires:
             - build
-      - docs-master:
```
Comment (ktabors) [inline .circleci/config.yml:177]: Should you also delete https://github.com/adobe-private/react-spectrum-v3/blob/cf5410adaedecec6a6dacfe1b7b7f556973bbbb2/.circleci/config.yml#L104-L121 ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -174,24 +174,12 @@ workflows:
...
-              only: master
-          requires:
-            - build
       - deploy:
           requires:
             - lint
             - test
             - storybook
             - docs
-      - deploy-master:
```
Comment (ktabors) [inline .circleci/config.yml:189]: Should you also delete https://github.com/adobe-private/react-spectrum-v3/blob/cf5410adaedecec6a6dacfe1b7b7f556973bbbb2/.circleci/config.yml#L135-L142 ?

File: .circleci/config.yml (hunk 1 above)
Comment (ktabors) [inline .circleci/config.yml:177]: Should you also delete https://github.com/adobe-private/react-spectrum-v3/blob/cf5410adaedecec6a6dacfe1b7b7f556973bbbb2/.circleci/config.yml#L104-L121 ?

Comment (snowystinger) [reply .circleci/config.yml:177]: not really any reason to, if devon doesn't end up needing it, we can delete it entirely

File: .circleci/config.yml (hunk 2 above)
Comment (snowystinger) [reply .circleci/config.yml:189]: not really any reason to

**Comment by dannify** [comment]
So I think we still want a way to deploy it. We just want to turn it off for now and figure out another trigger instead of on master merge

**Comment by snowystinger** [comment]
This was the fastest way to get back to being able to merge to master. And do we really want to be able to deploy the master docs like this in any case? If they're to be built against the deployed code, then we would never want these particular jobs to run. I didn't remove the regular PR builder, only the master one.

I may be misunderstanding though, in which case, what other trigger might you like? One's that are available are:
- `tags`, we could tag a PR as documentation only
- use a different job only on master to detect if it's documentation only, easy to just see if changes are only in certain directories, don't know if we'd want to do it for jsdocs changes as well as that'd be harder, eventually this will likely exist https://ideas.circleci.com/ideas/CCI-I-208
- we could have an explicit button that sends a post message with a parameter

**Comment by devongovett** [comment]
maybe just comment it out so we can bring it back easier?

**Pull request merged**